tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/testlaunch/LaunchShortcutUtil.xtend

```
@@ -0,0 +1,63 @@
+package org.testeditor.tcl.dsl.ui.testlaunch
+
+import javax.inject.Inject
+import org.apache.log4j.Logger
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IResource
+import org.eclipse.core.runtime.CoreException
+import org.eclipse.emf.common.util.URI
+import org.eclipse.jdt.core.IJavaElement
+import org.eclipse.jdt.core.IPackageFragment
+import org.eclipse.jdt.core.JavaCore
+import org.eclipse.xtext.resource.FileExtensionProvider
+import org.eclipse.xtext.ui.generator.IDerivedResourceMarkers
+
+class LaunchShortcutUtil {
+
+	val logger = Logger.getLogger(LaunchShortcutUtil)
+
+	@Inject IDerivedResourceMarkers derivedResourceMarkers;
+	@Inject FileExtensionProvider fileExtensionProvider;
+
+	/** is a valid object for executing a test on */
+	def boolean isValidForTestrun(Object res) {
+		res instanceof IResource && fileExtensionProvider.isValid((res as IResource).fileExtension)
+	}
+
+	/** get the derived java element of the given resource */
+	def IJavaElement getJavaElementForResource(IResource resource) {
+		try {
+			val sourcePath = URI.createPlatformResourceURI(resource.fullPath.toString, true).toString
+			val resources = derivedResourceMarkers.findDerivedResources(resource.project, sourcePath)
+			return JavaCore.create(resources.findFirst[containsElementsSearchedFor])
+		} catch (CoreException e) {
+			logger.debug(e.message, e);
```

method is allowed to fail silently returning null in that case.